sled-agent/config-reconciler/src/reconciler_task/zones.rs

```diff
@@ -0,0 +1,1111 @@
+            sled_agent_facilities
+                .start_omicron_zone(
+                    zone,
+                    &self.mount_config,
+                    is_time_synchronized,
+                    &all_u2_pools,
+                )
+                .map(move |result| {
+                    (
+                        zone.clone(),
+                        result.map_err(ZoneStartError::SledAgentStartFailed),
+                    )
+                })
+        });
+
+        // Concurrently start all zones, then record the results.
+        let start_results = future::join_all(start_futures).await;
+        for (config, result) in start_results {
+            let state = match result {
+                Ok(running_zone) => ZoneState::Running(Arc::new(running_zone)),
+                Err(err) => ZoneState::FailedToStart(err),
+            };
+            self.zones.insert(OmicronZone { config, state });
+        }
+    }
+
+    /// Check the timesync status from a running NTP zone (if it exists)
+    pub(super) async fn check_timesync(self) -> TimeSyncStatus {
+        match &self.timesync_config {
+            TimeSyncConfig::Normal => {
+                match self.timesync_status_from_ntp_zone().await {
+                    Ok(timesync) => TimeSyncStatus::TimeSync(timesync),
+                    Err(err) => {
+                        TimeSyncStatus::FailedToGetSyncStatus(Arc::new(err))
+                    }
+                }
+            }
+            TimeSyncConfig::Skip => TimeSyncStatus::ConfiguredToSkip,
+        }
+    }
+
+    async fn timesync_status_from_ntp_zone(
+        &self,
+    ) -> Result<TimeSync, TimeSyncError> {
+        // Get the one and only running NTP zone, or return an error.
+        let mut running_ntp_zones = self.zones.iter().filter_map(|z| {
+            if !z.config.zone_type.is_ntp() {
+                return None;
+            }
+
+            match &z.state {
+                ZoneState::Running(running_zone) => Some(running_zone),
+                ZoneState::PartiallyShutDown { .. }
+                | ZoneState::FailedToStart(_) => None,
+            }
+        });
+        let running_ntp_zone =
+            running_ntp_zones.next().ok_or(TimeSyncError::NoRunningNtpZone)?;
+        if running_ntp_zones.next().is_some() {
+            return Err(TimeSyncError::MultipleRunningNtpZones);
+        }
+
+        // XXXNTP - This could be replaced with a direct connection to the
+        // daemon using a patched version of the chrony_candm crate to allow
+        // a custom server socket path. From the GZ, it should be possible to
+        // connect to the UNIX socket at
+        // format!("{}/var/run/chrony/chronyd.sock", ntp_zone.root())
+
+        let stdout = running_ntp_zone
+            .run_cmd(&["/usr/bin/chronyc", "-c", "tracking"])
+            .map_err(TimeSyncError::ExecuteChronyc)?;
+
+        let v: Vec<&str> = stdout.split(',').collect();
```

Comment:
Would it make sense to have a `ntp_admin` server sort of like the ones we have for the clickhouse, keeper and cockroachdb zones? It'd be nice to be able to make an API call instead of zlogin-ing to the zone and running a command no?

Reply:
There's also the commented-out suggestion that we could run chronyc directly in the gz but point it to the socket in the NTP zone. This code is just moved as-is from sled-agent and I'd rather not change it as part of this PR, but yeah we should do better here.

I do think we'll definitely want an `ntp_admin` server when we get around to reworking RSS (e.g., to address #4587) - it'd be nice for wicket to expose some of the NTP debugging things we do by hand today (e.g., ping, resolving DNS), and that would probably be easiest via an `ntp_admin`-like thing.